placement/algo/sharded_helper.go

```diff
@@ -67,25 +67,25 @@ type PlacementHelper interface {
 	// HasRackConflict checks if the rack constraint is violated when moving the shard to the target rack.
 	HasRackConflict(shard uint32, from placement.Instance, toRack string) bool
 
-	// PlaceShards distributes shards to the instances in the helper, with aware of where are the shards coming from.
-	PlaceShards(shards []shard.Shard, from placement.Instance, candidates []placement.Instance) error
+	// placeShards distributes shards to the instances in the helper, with aware of where are the shards coming from.
```

Comment:
It's somewhat unconventional to have public interfaces with private APIs. If we don't want users to use these APIs, perhaps we should just remove them or move them to a private interface?

Reply:
Sounds good, separated the interfaces